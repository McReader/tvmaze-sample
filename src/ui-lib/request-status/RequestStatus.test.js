import renderer from "react-test-renderer";

import {
    REQUEST_STATUS_FULFILLED,
    REQUEST_STATUS_IDLE,
    REQUEST_STATUS_PENDING,
    REQUEST_STATUS_REJECTED,
    RequestStatus
} from "./RequestStatus";

let tree;
let props;

beforeEach(() => {
    tree = renderer
        .create(<RequestStatus {...props}>{ () => "Children" }</RequestStatus>)
        .toJSON();
});

describe(`when requestStatus "${REQUEST_STATUS_IDLE}"`, () => {
    beforeAll(() => {
        props = {
            requestStatus: REQUEST_STATUS_IDLE,
        };
    });

    test(`should render progress bar`, () => {
        expect(tree).toMatchSnapshot();
    });
});

describe(`when requestStatus "${REQUEST_STATUS_PENDING}"`, () => {
    beforeAll(() => {
        props = {
            requestStatus: REQUEST_STATUS_PENDING,
        };
    });

    test(`should render progress bar`, () => {
        expect(tree).toMatchSnapshot();
    });

    describe(`when requestStatus "${REQUEST_STATUS_REJECTED}"`, () => {
        beforeAll(() => {
            props = {
                requestStatus: REQUEST_STATUS_REJECTED,
            };
        });

        test(`should render caution symbol`, () => {
            expect(tree).toMatchSnapshot();
        });
    });

    describe(`when requestStatus "${REQUEST_STATUS_FULFILLED}"`, () => {
        beforeAll(() => {
            props = {
                requestStatus: REQUEST_STATUS_FULFILLED,
            };
        });

        test(`should render children`, () => {
            expect(tree).toMatchSnapshot();
        });
    });
});
