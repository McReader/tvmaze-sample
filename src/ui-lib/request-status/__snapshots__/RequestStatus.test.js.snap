// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when requestStatus "idle" should render progress bar 1`] = `
<div
  className="FillParentPlaceholder"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`when requestStatus "pending" should render progress bar 1`] = `
<div
  className="FillParentPlaceholder"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`when requestStatus "pending" when requestStatus "fulfilled" should render children 1`] = `"Children"`;

exports[`when requestStatus "pending" when requestStatus "rejected" should render caution symbol 1`] = `
<div
  className="FillParentPlaceholder"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
</div>
`;
